packages/core/src/messages/responses/item-type-api.ts

```diff
@@ -0,0 +1,79 @@
+import {SdkError} from '../../errors/errors';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseHit,
+  ResponseMiss,
+} from './response-base';
+import {cache_client} from '@gomomento/generated-types/dist/cacheclient';
```

Comment:
these responses should be independent of any grpc code. For example, since you are importing the nodejs types here, this will not run in the web browser. I think the constructor still should take in a string, not a grcp response type

Reply:
for example here https://github.com/momentohq/client-sdk-nodejs/blob/76b4b307feb8d197f503a294ab851486d302e900/packages/core/src/messages/responses/grpc-response-types.ts#L9-L14, we have our own enums that are independent of the enums inside of the protos, and these are the enums that the constructors are using https://github.com/momentohq/client-sdk-nodejs/blob/76b4b307feb8d197f503a294ab851486d302e900/packages/core/src/messages/responses/grpc-response-types.ts#L29

Reply:
vs using the enums defined in the proto. Allows us to be more flexible down the line

Reply:
yes, and in this case we should just define a public, first-class enum for these as part of this Response class.